src/test/java/com/axoniq/someproject/something/SomeAggregate.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.axoniq.someproject.something;
+
+import com.axoniq.someproject.api.ChangeStatusCommand;
+import com.axoniq.someproject.api.SomeCommand;
+import com.axoniq.someproject.api.SomeEvent;
+import com.axoniq.someproject.api.StatusChangedEvent;
+import org.axonframework.commandhandling.CommandHandler;
+import org.axonframework.eventsourcing.EventSourcingHandler;
+import org.axonframework.modelling.command.AggregateIdentifier;
+import org.axonframework.spring.stereotype.Aggregate;
+
+import java.util.Objects;
+
+import static org.axonframework.modelling.command.AggregateLifecycle.apply;
+
+@Aggregate
+public class SomeAggregate {
+
+    @AggregateIdentifier
+    private String id;
+    private String status;
+
+    @CommandHandler
+    public SomeAggregate(SomeCommand command) {
+        apply(new SomeEvent(command.id()));
+    }
+
+    @CommandHandler
+    public void handle(ChangeStatusCommand command) {
```

Comment:
Can we add some, or ideally all, parameters Axon Framework can resolve for every handler?

Reply:
I'm trying to understand why. We might, at some point, have an example app, to test the whole compilation to native and run the application. For now, the parameters seem to work, as long as reflection on the message handler methods is enabled.

Reply:
I figured that, especially for the `@Timestampe` and `@MetaDataValue` annotations, which we use on parameters, validating those would work in a test case makes sense.
Going from their to add the spring options with `@Autowired`/`@Required`/`@Qualifier` made sense to me to see if those work as intended too.

Reply:
They currently work, as it seems smart enough with the reflective hints on the methods that use them. That does, however make them hard to test with the current tests. It would be nice to add them to an and-to-and test in the future, so we can be sure they keep working.